src/gui/FileBrowser.cpp

```diff
@@ -162,17 +196,52 @@ bool FileBrowser::filterItems( const QString & filter, QTreeWidgetItem * item )
 }
 
 
+void filterDirectories(QString& dirstring, const QString& filterstring, bool checked)
+{
+	if (!checked)
+	{
+		dirstring = dirstring.replace(filterstring, "");
+	}
+	else
+	{
+		if (!dirstring.contains(filterstring))
+		{
+			dirstring += "*" + filterstring;
+		}
+	}
+	dirstring = dirstring.replace("**", "*");
+	if ((dirstring.length() > 0) && (*dirstring.begin() == '*'))
+	{
+		dirstring = dirstring.remove(0, 1);
+	}
+	if ((dirstring.length() > 0) && (*(dirstring.end()-1) == '*'))
+	{
+		dirstring = dirstring.remove(dirstring.length()-1, 1);
+	}
+}
+
 
 void FileBrowser::reloadTree( void )
 {
 	QList<QString> expandedDirs = m_fileBrowserTreeWidget->expandedDirs();
 	const QString text = m_filterEdit->text();
 	m_filterEdit->clear();
 	m_fileBrowserTreeWidget->clear();
-	QStringList paths = m_directories.split( '*' );
-	for( QStringList::iterator it = paths.begin(); it != paths.end(); ++it )
+	if ((m_showUserContent != nullptr) && (m_showFactoryContent != nullptr))
 	{
-		addItems( *it );
+		filterDirectories(m_directories, m_userDir,
+						  m_showUserContent->isChecked());
+		filterDirectories(m_directories, m_factoryDir,
+						  m_showFactoryContent->isChecked());
+	}
+
+	if (m_directories.length() > 0)
+	{
+		QStringList paths = m_directories.split( '*' );
+		for( QStringList::iterator it = paths.begin(); it != paths.end(); ++it )
+		{
+			addItems( *it );
```

Comment:
I still didn't get to test the build (will download it later), but as far as the code goes, it all looks good to me. Both the logic and the code style of new additions (except for this little section here). I'll update my review to an approval as soon as I test it!

```suggestion
		QStringList paths = m_directories.split('*');
		for (QStringList::iterator it = paths.begin(); it != paths.end(); ++it)
		{
			addItems(*it);
```